File: Java/LeetCode/Rings and Rods.java
https://leetcode.com/problems/rings-and-rods/

class Solution {

    public int countPoints(String rings) {
        Map<Integer, Set<Character>> rodColors = new HashMap<>();

        for (int i = 0; i < rings.length()-1; i += 2) {
            char ringColor = rings.charAt(i);
            int rodNumber = Character.getNumericValue( rings.charAt(i+1) );

            Set<Character> currentRodColors = rodColors.getOrDefault(rodNumber, new HashSet<>());
            currentRodColors.add(ringColor);
            rodColors.put(rodNumber, currentRodColors);
        }

        int result = 0;
        for (Set<Character> currentRodColors : rodColors.values()) {
            if (currentRodColors.size() == 3) {
                result++;
            }
        }

        return result;
    }
    
}

File: Kotlin/LeetCode/Rings and Rods.kt
// https://leetcode.com/problems/rings-and-rods/

class Solution {

    fun countPoints(rings: String): Int {
        val rodColors = mutableMapOf<Int, MutableSet<Char>>();

        for (i in 0 until rings.length-1 step 2) {
            val ringColor = rings[i]
            val rodNumber = rings[i+1].toInt()

            val currentRodColors = rodColors.getOrDefault(rodNumber, mutableSetOf<Char>());
            currentRodColors.add(ringColor);
            rodColors.put(rodNumber, currentRodColors);
        }

        var result = 0;
        for (currentRodColors in rodColors.values) {
            if (currentRodColors.size == 3) {
                result++;
            }
        }

        return result;
    }

}
